pkg/plugins/manager/manager_test.go

```diff
@@ -221,17 +289,123 @@ func TestPluginManager_IsBackendOnlyPlugin(t *testing.T) {
 	}
 }
 
+func TestPluginManager_Installer(t *testing.T) {
+	t.Run("Install plugin after manager init", func(t *testing.T) {
+		fm := &fakeBackendPluginManager{}
+		pm := createManager(t, func(pm *PluginManager) {
+			pm.BackendPluginManager = fm
+		})
+
+		err := pm.Init()
+		require.NoError(t, err)
+
+		// mock installer
+		installer := &fakePluginInstaller{}
+		pm.pluginInstaller = installer
+
+		// Set plugin location (we do this after manager Init() so that
+		// it doesn't install the plugin automatically)
+		pm.Cfg.PluginsPath = "testdata/installer"
+
+		pluginID := "test"
+		pluginFolder := pm.Cfg.PluginsPath + "/plugin"
+
+		err = pm.Install(context.Background(), pluginID, "1.0.0")
+		require.NoError(t, err)
+
+		assert.Equal(t, 1, installer.installCount)
+		assert.Equal(t, 0, installer.uninstallCount)
+
+		// verify plugin manager has loaded core plugins successfully
+		assert.Empty(t, pm.scanningErrors)
+		assert.Len(t, pm.Plugins(), 45)
+		assert.Len(t, pm.DataSources(), 21)
+		assert.Len(t, pm.Panels(), 24)
+		assert.Len(t, pm.Apps(), 0)
+
+		// verify plugin has been loaded successfully
+		assert.NotNil(t, pm.plugins[pluginID])
+		assert.Equal(t, "datasource", pm.plugins[pluginID].Type)
+		assert.Equal(t, "Test", pm.plugins[pluginID].Name)
+		assert.Equal(t, pluginID, pm.plugins[pluginID].Id)
+		assert.Equal(t, "1.0.0", pm.plugins[pluginID].Info.Version)
+		assert.Equal(t, plugins.PluginSignatureValid, pm.plugins[pluginID].Signature)
+		assert.Equal(t, plugins.GrafanaType, pm.plugins[pluginID].SignatureType)
+		assert.Equal(t, "Grafana Labs", pm.plugins[pluginID].SignatureOrg)
+		assert.Equal(t, pluginFolder, pm.plugins[pluginID].PluginDir)
+		assert.False(t, pm.plugins[pluginID].IsCorePlugin)
```

Comment:
I would recommend using go-cmp here, for a more comprehensive equality assertion.